arch/rv32i/src/lib.rs

```diff
@@ -15,6 +15,11 @@ pub mod support;
 pub mod syscall;
 extern crate tock_registers;
 
+#[cfg(not(all(feature = "riscv64", target_os = "none")))]
+pub const XLEN: usize = 32;
+#[cfg(feature = "riscv64")]
+pub const XLEN: usize = 64;
+
```

Comment:
In Tock we generally try to avoid `#[cfg]` when possible. Could this be replaced with `pub const XLEN: usize = core::mem::size_of::<usize>() * 8;`?